lib/collection/src/recommendations.rs

```diff
@@ -27,14 +25,14 @@ use crate::operations::types::{
     RecommendRequestInternal, RecommendStrategy, UsingVector,
 };
 
-fn avg_vectors<'a>(vectors: impl Iterator<Item = VectorRef<'a>>) -> CollectionResult<Vector> {
+fn avg_vectors(vectors: impl Iterator<Item = Vector>) -> CollectionResult<Vector> {
```

Comment:
Downside of this PR. We take ownership here because `VectorRef` is an segment internal structure not related to `Vector` anymore